sale_force_invoiced_qty/models/sale_order_line.py

```diff
@@ -0,0 +1,33 @@
+# Copyright (C) 2024 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    force_invoiced = fields.Float()
+
+    @api.depends("force_invoiced")
+    def _compute_qty_to_invoice(self):
+        """
+        Compute the quantity to invoice.
+        """
+        res = super()._compute_qty_to_invoice()
+        for line in self:
+            if line.state in ["sale", "done"] and not line.display_type:
+                if line.product_id.invoice_policy != "order":
+                    qty_to_invoice = (
+                        line.qty_delivered - line.qty_invoiced - line.force_invoiced
+                    )
+                    if qty_to_invoice < 0:
```

Comment:
This is a good idea to add this check. However let's remove this check for now in case user wants to set negative qty on purpose.